adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/ADLSBatchSink.java

```diff
@@ -128,18 +151,60 @@ public void transform(StructuredRecord input, Emitter<KeyValue<AvroKey<GenericRe
     @Macro
     public String fileSystemProperties;
 
-    @Description("Output schema for the output files.")
+    @Description("The format of output files.")
+    @Macro
+    private String outputFormat;
+
+    @Nullable
+    @Description("Output schema of the JSON document. Required for avro output format. " +
+      "If left empty for text output format, the schema of input records will be used." +
+      "This must be a subset of the schema of input records. " +
+      "Fields of schema types ARRAY, MAP, RECORD and UNION in input records cannot be present " +
+      "in output schema of text output format.")
     @Macro
     public String schema;
 
+    @Nullable
+    @Macro
+    @Description("Field delimiter for text format output files. Defaults to tab.")
+    public String fieldDelimiter;
+
     public AzureBatchSinkConfig(String referenceName) {
       super(referenceName);
     }
 
+    protected void validate(Schema inputSchema) {
+      validate();
+      if (!containsMacro("schema")) {
+        for (Schema.Field outputField : getSchema().getFields()) {
+          String fieldName = outputField.getName();
+          Schema.Field inputField = inputSchema.getField(outputField.getName());
+          if (inputField == null) {
+            throw new IllegalArgumentException("Input record does not contain the " + fieldName
+                                                 + " field in output schema.");
+          }
+          if (TEXT.equals(outputFormat)) {
+            Schema.Type inputType = inputField.getSchema().getType();
+            switch (inputType) {
+              case ARRAY:
+              case MAP:
+              case RECORD:
+              case UNION:
```

Comment:
nullables are unions. This should only be checked if the schema is a union that does not represent a nullable type.